lib/iris/tests/unit/lazy_data/test_map_complete_blocks.py

```diff
@@ -134,3 +135,24 @@ def test_multidimensional_input(self):
         )
         assert is_lazy_data(result)
         assert_array_equal(result.compute(), array + 1)
+
+    def test_rechunking(self):
+        # Choose a dask array with an irregularly chunked dimension to be rechunked.
+        lazy_array = da.ones((5, 9, 10, 10), chunks=(2, 5, 10, 5))
+        cube, _ = create_mock_cube(lazy_array)
+
+        result = map_complete_blocks(
+            cube, self.func, dims=(2, 3), out_sizes=(30, 40), dtype=lazy_array.dtype
+        )
+        assert is_lazy_data(result)
+        # Reduce the optimum dask chunksize.
+        with dask.config.set({"array.chunk-size": "32KiB"}):
+            result = map_complete_blocks(
+                cube, self.func, dims=(2, 3), out_sizes=(30, 40), dtype=lazy_array.dtype
```

Comment:
I think it would be a more convincing example if the "fixed" dims weren't at the end of the shape.

I think you can easily transpose the example so the initial content of
`da.ones((5, 9, 10, 10), chunks=(2, 5, 10, 5))`
 ==> `da.ones((5, 10, 9, 10), chunks=(2, 10, 5, 5))`
 And here in the map call we use `dims=(1, 3)`.
 
 (N.B. I did try this, and it does actually seem to work!)

Reply:
Nice,thanks !